test/driver_ordering_helper_tests.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ *
+ * Copyright (C) 2025 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#include "gtest/gtest.h"
+
+// To test internal functions, we need to include the implementation
+// In a real scenario, these functions might need to be exposed via a test interface
+// or moved to a separate testable module
+
+#include <string>
+#include <vector>
+
+// Include the loader types we need
+#include "source/loader/ze_loader_internal.h"
+
+// Since the helper functions are currently static/internal, we'll redefine them here for testing
+// In a production environment, these would ideally be extracted to a testable module
+
+namespace loader_test {
+
+// Copy of the internal helper functions for testing purposes
+enum DriverOrderSpecType { BY_GLOBAL_INDEX, BY_TYPE, BY_TYPE_AND_INDEX };
+
+struct DriverOrderSpec {
+    DriverOrderSpecType type;
+    uint32_t globalIndex = 0;
+    loader::zel_driver_type_t driverType = loader::ZEL_DRIVER_TYPE_FORCE_UINT32;
+    uint32_t typeIndex = 0;
+};
+
+// Helper function to map driver type string to enum (copy from implementation)
+loader::zel_driver_type_t stringToDriverType(const std::string& typeStr) {
+    if (typeStr == "DISCRETE_GPU_ONLY") {
+        return loader::ZEL_DRIVER_TYPE_DISCRETE_GPU;
+    } else if (typeStr == "GPU") {
+        return loader::ZEL_DRIVER_TYPE_GPU;
+    } else if (typeStr == "INTEGRATED_GPU_ONLY") {
+        return loader::ZEL_DRIVER_TYPE_INTEGRATED_GPU;
+    } else if (typeStr == "NPU") {
+        return loader::ZEL_DRIVER_TYPE_NPU;
+    }
+    return loader::ZEL_DRIVER_TYPE_FORCE_UINT32; // Invalid
+}
+
+// Helper function to trim whitespace (copy from implementation)
```

Comment:
In the future, it would make more sense to have a ze_loader_util.h or something with these things defined, since we're duplicating functions between code bodies.

Reply:
fixed in latest